psql/psqltest/psqltest.go

```diff
@@ -0,0 +1,79 @@
+package psqltest
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"go.soon.build/kit/psql"
+)
+
+func createTestDb(name string, db *sql.DB) error {
+	defer db.Close()
+	_, err := db.Exec(fmt.Sprintf(`DROP DATABASE IF EXISTS "%s"`, name))
+	if err != nil {
+		return fmt.Errorf("error dropping existing test database: %v", err)
+	}
+	_, err = db.Exec(fmt.Sprintf(`CREATE DATABASE "%s"`, name))
+	if err != nil {
+		return fmt.Errorf("error creating test database: %v", err)
+	}
+	return nil
+}
+
+// Open connects to the default database
+// then creates a new database from the test name
+func Open(ctx context.Context, t *testing.T, dbCfg psql.Config) *sql.DB {
+	// Create test database
+	db, err := psql.Open(ctx, dbCfg)
+	if err != nil {
+		t.Fatalf("err opening test db: %v", err)
+	}
+	testDbName := t.Name()
+	t.Logf("creating test db: %s", testDbName)
+	err = createTestDb(testDbName, db)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Open test database
+	tdb, err := psql.Open(ctx, psql.Config{
+		User:    dbCfg.User,
+		Pass:    dbCfg.Pass,
+		Host:    dbCfg.Host,
+		SSLMode: dbCfg.SSLMode,
+		DBName:  testDbName,
+	})
```

Comment:
you could simplify this as its the same type as dbCfg
```
dbCfg. DBName = t.Name()
tdb, err := psql.Open(ctx, dbCfg)
```
there are no worries about overwriting the config passed in as its a value not a pointer